src/_store/AppStateProvider.js

```diff
@@ -10,6 +10,18 @@ import { appStateSelector } from '../_selectors/AppStateSelectors';
 @connect(appStateSelector)
 export default class AppStateProvider extends Component {
 
+    componentWillMount() {
+        this.setState({ timer: window.setTimeout(this.showMessageForSlowConnection.bind(this), 4000) });
+    }
+
+    componentWillUnmount() {
+        window.clearTimeout(this.state.timer);
```

Comment:
No need for state. This from clock component:

componentDidMount() {
        this.interval = setInterval(() =>
            this.setState({ time: Date.now() }), 1000);
    }

    componentWillUnmount() {
		clearInterval(this.interval);
	}